fixes behav in singl-session-subjects in long study

fliem opened the pull request:
In longitudinal studies (studies in which at least one subject has more than one session) and  `--multiple_sessions='longitudinal')` subjects with only one session should undergo the entire longitudinal stream. The changes in #11 broke that behavior.
The present changes should fix it.

fliem committed:
fixes behav in singl-session-subjects in long study

alexlicohen reviewed (changes requested):
this change restarts the for `subject_label` loop after `session_dirs` and `sessions` is set on a per subject basis (line 111)

would just changing the > sign (line 120) to >= accomplish the same thing?

fliem committed:
fixes session var in second subject loop

fliem committed:
added longitudinal tests. tests now check for recon-all.done file

fliem commented:
Thanks. We need two loops here. The first determines if the study is longitudinal. For that it needs to loop through all subjects and look for sessions. The second loop starts the recon-all. 
* But you're right, I've missed the sessions variables. I've added them.
* I also added longitudinal tests & made the tests more thorough by checking if a recon-all.done file exists.

fliem committed:
missing cat

fliem committed:
removed longitudinal tests as they time-out (>120min)

fliem commented:
There still is something wrong here. I'll have a go at it again in the next days.
